### PR TITLE
fix typo; batch executemany in updateInputFilesStagedAboutIdds

### PR DESCRIPTION
**Pull request by mightqxc**

### CODE REVIEW

**Comment by wguanicedew** [comment]
I cannot install this branch. I tried to install it with pip. However, it requires panda-common>=0.1.3. 
I tried to install panda-common 'refs/pull/53/head' at first, and then install jedi without '--force'. In this way it doesn't install the jedi codes. If I install it with '--force', it will require panda-common>=0.1.3 even I already installed 'panda-common>=0.1.3' manually.

```
pip install --force git+https://github.com/PanDAWMS/panda-jedi.git@refs/pull/283/head
Collecting git+https://github.com/PanDAWMS/panda-jedi.git@refs/pull/283/head
  Cloning https://github.com/PanDAWMS/panda-jedi.git (to revision refs/pull/283/head) to /tmp/pip-req-build-if4_x896
  Running command git clone --filter=blob:none --quiet https://github.com/PanDAWMS/panda-jedi.git /tmp/pip-req-build-if4_x896
  WARNING: Did not find branch or tag 'refs/pull/283/head', assuming revision or ref.
  Running command git fetch -q https://github.com/PanDAWMS/panda-jedi.git refs/pull/283/head
  Running command git checkout -q e18292a80acb63525a0829c781d343d42f332d08
  Resolved https://github.com/PanDAWMS/panda-jedi.git to commit e18292a80acb63525a0829c781d343d42f332d08
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy (from panda-jedi==0.4.4)
  Using cached numpy-2.1.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (62 kB)
Collecting packaging (from panda-jedi==0.4.4)
  Using cached packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
INFO: pip is looking at multiple versions of panda-jedi to determine which version is compatible with other requirements. This could take a while.
ERROR: Ignored the following versions that require a different python version: 1.21.2 Requires-Python >=3.7,<3.11; 1.21.3 Requires-Python >=3.7,<3.11; 1.21.4 Requires-Python >=3.7,<3.11; 1.21.5 Requires-Python >=3.7,<3.11; 1.21.6 Requires-Python >=3.7,<3.11
ERROR: Could not find a version that satisfies the requirement panda-common>=0.1.3 (from panda-jedi) (from versions: 0.0.12, 0.0.14, 0.0.15, 0.0.16, 0.0.17, 0.0.18, 0.0.19, 0.0.20, 0.0.21, 0.0.24, 0.0.26, 0.0.27, 0.0.28, 0.0.29, 0.0.31, 0.0.32, 0.0.33, 0.0.34, 0.0.35, 0.0.36, 0.0.41, 0.0.42, 0.0.43, 0.0.44, 0.0.45, 0.1.0, 0.1.1, 0.1.2)
ERROR: No matching distribution found for panda-common>=0.1.3
```
```
pip install --force git+https://github.com/PanDAWMS/panda-common.git@refs/pull/53/head
```

**Comment by mightqxc** [comment]
Hi @wguanicedew 
Try with `--no-deps` i.e. 
`pip install --no-deps --force-reinstall git+https://github.com/PanDAWMS/panda-jedi.git@refs/pull/283/head`
I will try to merge panda-common v0.1.3 asap

File: pandajedi/jedicore/JediDBProxy.py [hunk 1]
```diff
@@ -13622,49 +13623,52 @@ def updateInputFilesStagedAboutIdds_JEDI(self, jeditaskid, scope, filenames_dict
...
-                    tmpLog.debug(f"running sql executemany: {sqlUF_with_fileID}")
-                    self.cur.executemany(sqlUF_with_fileID + comment, varMaps)
-                    retVal += self.cur.rowcount
+                    for one_batch in batched(filenames_dict_with_fileID.items(), chunk_size):
+                        # loop batches of executemany
+                        varMaps = []
+                        for filename, (datasetid, fileid) in one_batch:
+                            tmp_varMap = varMap.copy()
+                            tmp_varMap[":fileID"] = fileid
+                            del tmp_varMap[":scope"]
```
Comment (wguanicedew) [inline pandajedi/jedicore/JediDBProxy.py:13632]: it needs to check whether ':scope' is in the dict. when scope == "pseudo_dataset", ':scope' is not in the dict.
if ":scope" in tmp_varMap:
    del tmp_varMap[":scope"]

**Comment by wguanicedew** [comment]
I tried it with Tadashi's new panda-server updates and tested with 2 big workflow. I don't see this error. I didn't do big scale tests since the site is reserved for production processing. However, I think we can merge this one.

**Pull request merged**